src/mongocrypt-ctx-datakey.c

```diff
@@ -63,17 +63,11 @@ _kms_start (mongocrypt_ctx_t *ctx)
    _mongocrypt_kms_ctx_cleanup (&dkctx->kms);
    memset (&dkctx->kms, 0, sizeof (dkctx->kms));
    dkctx->kms_returned = false;
-   if (ctx->opts.masterkey_kms_provider == MONGOCRYPT_KMS_PROVIDER_LOCAL) {
+   if (ctx->opts.kek.kms_provider == MONGOCRYPT_KMS_PROVIDER_LOCAL) {
       bool crypt_ret;
       uint32_t bytes_written;
       _mongocrypt_buffer_t iv;
 
-      if (ctx->opts.masterkey_aws_endpoint) {
-         _mongocrypt_ctx_fail_w_msg (
-            ctx, "endpoint not supported for local masterkey");
-         goto done;
-      }
```

Comment:
Because the endpoint is part of the `_mongocrypt_kek_t.provider` union, it cannot be set independently of other AWS options. There's no need to check, since setting the endpoint after configuring a different KEK KMS provider is now an error.